### ISSUE TITLE
Not all icons are coming through (elementary OS Juno, elementary icons)

### ISSUE THREAD

**Issue by wout** [ebeem/inkscape-elegant#1]
First of all, thank you for this. It's really convenient.

I haven an issue, though. Not all icons are coming through. Would you know why?

![image](https://user-images.githubusercontent.com/107324/47608873-01049700-da35-11e8-8724-b1cc335cafeb.png)

I'm on elementary OS Juno.

**Comment by ebeem** [comment]
thanks for trying this out!!
the problem seems to be with your icons theme

can you let me know what icons-theme you use?
also, did you move the new generated icons as suggested?

the problem with moving the icons to hicolor or other icons is that the current icon theme will overwrite any icon it has ... and that's what I think your problem is ...
since you probably don't wanna mess up your icon theme
I suggest you do the following

you copy your current icons theme, and then generate new icons from the python tool and let them overwrite you copied icons theme. Use your new copied icons theme by using gnome-tweak or something

**Comment by wout** [comment]
Thanks for your quick reply. I'm on elementary OS Juno. You are right, the system is overwriting the custom icons where it has them available. I found them in `/usr/share/icons/elementary/actions/24`, so I copied `24` to `24.old` and threw in all icons generate by your script.

But, some icons are not coming through:

![image](https://user-images.githubusercontent.com/107324/47618543-c5210e80-dad4-11e8-9eca-b75eedb5c066.png)

It's a bit of an icon mess on Linux :)

**Comment by ebeem** [comment]
I am downloading elementary-os
will check it and let you know what happens ...

there are some icons that are not symbolized yet, but most of them are
the aim of this project is to redesign them all anyways

thanks

**Comment by wout** [comment]
Great, thanks! Just FYI, I installed Inkscape using `apt`. Not sure if that makes any difference compared to `snap` for example.

**Comment by ebeem** [comment]
not at all ... I am pretty much sure the problem is with the icons theme
actually I have a better idea to improve the script a bit ...
it's by creating a new icons theme that inherits your current one ... and then setting it as the current theme
after adding the new generated icons to it 

 if this works, it will be great

**Comment by ebeem** [comment]
ok, so I managed to do it ... and yes, it's the icon-theme problem
good thing is, we use scalable in this project ... so for now, you will not be overwriting/replacing any existing icons in your elementary theme. however, these new Inkscape icons will be applied elsewhere when they are needed.

I fixed the problem by introducing the scalable/actions directory to the current theme
 I did this first to generate the new icons to a new folder in the theme called scalable

`sudo python change_colors.py black scalable /usr/share/icons/elementary/scalable`

next, I edited the file `/usr/share/icons/elementary/index.theme` and applied these steps
1- go to the line that starts with `Directories=` (line#7) and add a new directory `scalable/actions` 
2- add this new section after the `Directories=` line 
```
[scalable/actions]
MinSize=8
Size=16
MaxSize=512
Context=Actions
Type=Scalable
```

3- re-apply your theme (go to gnome-tweaks, select another theme and select the `elementary` again)

of course I will be looking into a better solution that does not need user involvment and that's independant of the theme ..

let me know if this works for you

**Comment by wout** [comment]
Now it seems to be the other way around. 😂

![image](https://user-images.githubusercontent.com/107324/47620517-379ce900-daeb-11e8-8da2-7900f37922aa.png)

In the toolbar at least. Because on the right there still are some of the standard icons showing, if you compare the two screenshots.

**Comment by ebeem** [comment]
whoops, that's weird
here are easier steps, I just tried to reproduce .. it's working
1- `sudo python change_colors.py black scalable /usr/share/icons/elementary/scalable`
2- replace the content of `/usr/share/icons/elementary/index.theme` with this
https://gist.github.com/ebeem/ab21798890717f7814078f3c6b898567

of course re-apply the theme after

**Comment by wout** [comment]
That's more like it: 😎

![image](https://user-images.githubusercontent.com/107324/47620789-bd6e6380-daee-11e8-8742-7d7dd3a4ace7.png)

Thanks!

**Comment by ebeem** [comment]
glad it worked out!!!
I will close this for now, and I will surely implement a better way to apply the icons soon
let me know if you have any other problems

**Comment by ebeem** [comment]
I am happy to tell you that I have updated the way the tool works 
with the new optional argument ```--create-theme``` you can easily let the tool create a new icon-theme for you that inherits your current one, moves the generated icons to it, and apply it

this will give the advantages of not needing to touch your original icon-theme and having a simple icon theme where all inkscape-only-related icons are located

you can check the readme for more information

**Comment by wout** [comment]
Works great, thanks! Although, one small issue here where the elementary system icon for recent files is replaced with one from the Inkscape theme:

![image](https://user-images.githubusercontent.com/107324/47707200-82be1580-dc2b-11e8-815d-d56a4ac8c66c.png)

**Comment by ebeem** [comment]
I am afraid that it's difficult to escape this issue ... as Inkscape uses system icon-theme which is a modified version of elementary with some icons overwritten.

you can use gnome-tweaks to set the theme to elementary if it bothers you.
I will try to look into this, I have an idea in mind ... we probably can edit the shortcut of Inkscape to make it use a custom icon-theme, so I am thinking about.

will let you know if it works out.

Thanks for reporting

**Comment by wout** [comment]
That would be great, because a lot of other apps are affected as well:

![image](https://user-images.githubusercontent.com/107324/47717082-fd475f00-dc44-11e8-8396-358f0f40ae7c.png)

Almost every single native elementary apps have at lease one icon replaced, sometimes very unfortunate with dark themes enabled:

![image](https://user-images.githubusercontent.com/107324/47717166-50211680-dc45-11e8-970a-bcc0bff36a6f.png)

**Comment by wout** [comment]
That would be great, because a lot of other apps are affected as well:

![image](https://user-images.githubusercontent.com/107324/47717082-fd475f00-dc44-11e8-8396-358f0f40ae7c.png)

Almost every single native elementary apps have at lease one icon replaced, sometimes very unfortunate with dark themes enabled:

![image](https://user-images.githubusercontent.com/107324/47717166-50211680-dc45-11e8-970a-bcc0bff36a6f.png)

**Comment by ebeem** [comment]
sadly, I could NOT find any environment variable which helps to achieve such behavior...
setting GTK theme per application is possible by using the variable `GTK2_RC_FILES`
I tried a custom theme and it worked fine, but when I tried to set the icons via the RC property
`gtk-icon-theme-name = "Adwaita"`
nothing was affected, I might need help on this.
currently, can't see a way other than Inkscape supporting icons theme picking (same as LibreOffice & gimp)

However, I will try to look for workarounds and let you know if I find anything useful.
this issue will only affect Linux users.
